packages/core/src/Cardano/types/Cip1854PublicKey.ts

```diff
@@ -0,0 +1,55 @@
+import * as BaseEncoding from '@scure/base';
+import { InvalidStringError, OpaqueString, assertIsBech32WithPrefix, assertIsHexString } from '@cardano-sdk/util';
+
+const MAX_BECH32_LENGTH_LIMIT = 1023;
+export type Cip1854PublicKey = OpaqueString<'Cip1854PublicKey'>;
+
+export const Cip1854PublicKey = (value: string): Cip1854PublicKey => {
+  try {
+    assertIsBech32WithPrefix(value, ['addr_shared_vk', 'stake_shared_vk']);
+  } catch {
+    throw new InvalidStringError(value, 'Expected key to be a bech32 encoded string');
+  }
+
+  return value as Cip1854PublicKey;
+};
+
+/**
+ * Encodes a hex 1854 public key according to CIP5 https://cips.cardano.org/cip/CIP-5/ specification.
+ *
+ * @param value The hex value.
+ * @param prefix the prefix that matches CIP5 spec.
+ * @throws InvalidStringError if the value is not a valid hex string.
+ * @returns The bech32 encoded public key.
+ */
+const toBech32 = (value: string, prefix: string): Cip1854PublicKey => {
+  try {
+    assertIsHexString(value);
+  } catch {
+    throw new InvalidStringError(value, 'Expected key to be a hex-encoded bytes');
+  }
+
+  const words = BaseEncoding.bech32.toWords(Buffer.from(value, 'hex'));
+  return Cip1854PublicKey(BaseEncoding.bech32.encode(prefix, words, MAX_BECH32_LENGTH_LIMIT));
+};
+
+/**
+ * Decodes a bech32 string to hex value.
+ *
+ * @param value A Cip1854PublicKey value to convert.
+ * @returns A hex string.
+ */
+const fromBech32 = (value: Cip1854PublicKey): string => {
+  const { words } = BaseEncoding.bech32.decode(value, MAX_BECH32_LENGTH_LIMIT);
+  return Buffer.from(BaseEncoding.bech32.fromWords(words)).toString('hex');
+};
+
+Cip1854PublicKey.Stake = {
```

Comment:
I think its best to have two separate types, one for the payment key and one for the staking key:

Cip1854PaymentPublicKey and Cip1854StakePublicKey